app/lib/dns.server.ts

```diff
@@ -0,0 +1,193 @@
+      throw new Error(
+        `Error occurred while creating resource record: ${error.message}`
+      );
+    }
+  }
+};
+
+export const upsertRecord = async (
+  hostedZoneId: string,
+  type: RecordType,
+  name: string,
+  value: string
+): Promise<string | undefined> => {
+  try {
+    if (!isNameValid(name)) {
+      throw new Error("Invalid name provided");
+    }
+
+    if (!isValueValid(type, value)) {
+      throw new Error("Invalid value provided");
+    }
+
+    const command = new ChangeResourceRecordSetsCommand({
+      ChangeBatch: {
+        Changes: [
+          {
+            Action: "UPSERT",
+            ResourceRecordSet: {
+              Name: name,
+              Type: type,
+              ResourceRecords: [
+                {
+                  Value: value,
+                },
+              ],
+            },
+          },
+        ],
+      },
+      HostedZoneId: hostedZoneId,
+    });
+    const response = await route53Client.send(command);
+
+    if (!response.ChangeInfo?.Id) {
+      throw new Error(`Missing ID in ChangeInfo`);
+    }
+    return response.ChangeInfo?.Id;
+  } catch (error) {
+    if (error instanceof Error) {
+      throw new Error(
+        `Error occurred while updating resource record: ${error.message}`
+      );
+    }
+  }
+};
+
+export const deleteRecord = async (
+  hostedZoneId: string,
+  type: RecordType,
+  name: string,
+  value: string
+): Promise<string> => {
+  try {
+    if (!isNameValid(name)) {
+      throw new Error("Invalid name provided");
+    }
+
+    if (!isValueValid(type, value)) {
+      throw new Error("Invalid value provided");
+    }
+
+    const command = new ChangeResourceRecordSetsCommand({
+      ChangeBatch: {
+        Changes: [
+          {
+            Action: "DELETE",
+            ResourceRecordSet: {
+              Name: name,
+              Type: type,
+              ResourceRecords: [
+                {
+                  Value: value,
+                },
+              ],
+            },
+          },
+        ],
+      },
+      HostedZoneId: hostedZoneId,
+    });
+
+    const response = await route53Client.send(command);
+
+    if (!response.ChangeInfo?.Id) {
+      throw new Error("Missing ID in ChangeInfo");
+    }
+    return response.ChangeInfo?.Id;
+  } catch (error) {
+    throw new Error(`Error occurred while deleting resource record: ${error}`);
+  }
+};
+
+//requestId to changeId?
+export const getChangeStatus = async (changeId: string) => {
+  try {
+    const command = new GetChangeCommand({
+      Id: changeId,
+    });
+    const response = await route53Client.send(command);
+
+    if (!response.ChangeInfo?.Status) {
+      throw new Error("Could not get ChangeIno for requested ID");
+    }
+    return response.ChangeInfo?.Status;
+  } catch (error) {
+    throw new Error(`Error occurred while getting change status: ${error}`);
+  }
+};
+
+const isNameValid = (name: string) => {
+  const subdomain = name.replace(".starchart.com", "");
```

Comment:
We shouldn't have anything hard-coded in here.  We'll test the entire name (i.e., get rid of this line).